packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -111,12 +113,50 @@ class ColdRunner extends ResidentRunner {
     return 0;
   }
 
+  @override
+  Future<int> attach({
+    Completer<DebugConnectionInfo> connectionInfoCompleter,
+    Completer<void> appStartedCompleter,
+  }) async {
+    _didAttach = true;
+    try {
+      await connectToServiceProtocol();
+    } catch (error) {
+      printError('Error connecting to the service protocol: $error');
+      return 2;
```

Comment:
what does returning 2 mean?

Reply:
I believe it is a signal used by the daemon. I will find the source and give this and the version in run_hot a doc comment

Reply:
I copied it from run_hot attach, maybe you know from https://github.com/flutter/flutter/pull/8764 ? It doesn't seem to go anywhere

Reply:
I probably copied it from somewhere as well back then, unfortunately.